build/publisher.go

```diff
@@ -22,32 +24,27 @@ type myMux struct {
 }
 
 func (m *myMux) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+
+	logger := hlog.FromRequest(req)
 	// File to server
 	baseURL := m.baseURL
 	documentRoot := m.documentRoot
 
-	userAgent := req.UserAgent()
-
-	xff := req.Header.Get("X-FORWARDED-FOR")
-	remoteAddr := req.RemoteAddr
-
-	var requestor string
-	if len(xff) > 0 {
-		requestor = xff
-	} else if len(remoteAddr) > 0 {
-		splitAddr := strings.Split(remoteAddr, ":")
-		requestor = splitAddr[0]
-	} else {
-		requestor = "UNKNOWN"
+	xffs := req.Header["X-Forwarded-For"]
+	if len(xffs) > 0 {
+		// From HAProxy's documentation:
+		// Since this header is always appended at the end of the existing header list, the server must be configured to always use the last occurrence of this header only.
+		xff := xffs[len(xffs)-1]
+		logger.UpdateContext(func(c zerolog.Context) zerolog.Context {
+			return c.Str("x_forwarded_for", xff)
```

Comment:
Nice, an even more nice sollution would be to use the scheme mozilla suggests here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
ie. join all x-forwarded-for with a comma.

Reply:
It's doable. But do we need it?

I don't think we should cover it in this PR.